velox/common/caching/SsdFile.cpp

```diff
@@ -583,9 +583,14 @@ bool SsdFile::removeFileEntries(
   }
   if (toFree.size() > 0) {
     clearRegionEntriesLocked(toFree);
-    writableRegions_.reserve(writableRegions_.size() + toFree.size());
+    writableRegions_.reserve(std::min(
+        writableRegions_.size() + toFree.size(),
+        (std::vector<int>::size_type)numRegions_));
     for (int32_t region : toFree) {
-      writableRegions_.push_back(region);
+      if (std::find(writableRegions_.begin(), writableRegions_.end(), region) ==
```

Comment:
Can we setup a temporal folly unordered set of existing writableRegions_  for this check? We might have many regions? Thanks!